twistr/space_race.go

```diff
@@ -0,0 +1,67 @@
+package twistr
+
+import "fmt"
+
+type SRId int
+
+const (
+	NoAbility SRId = iota
+	TwoSpace
+	OppHeadlineFirst
+	DiscardHeld
+	ExtraAR
+)
+
+type SRBox struct {
+	MaxRoll    int
+	OpsNeeded  int
+	FirstVP    int
+	SecondVP   int
+	SideEffect SRId
+}
+
+var SRTrack []SRBox = []SRBox{
+	SRBox{0, 0, 0, 0, NoAbility},
+	SRBox{3, 2, 2, 1, NoAbility},
+	SRBox{4, 2, 0, 0, TwoSpace},
+	SRBox{3, 2, 2, 0, NoAbility},
+	SRBox{4, 2, 0, 0, OppHeadlineFirst},
+	SRBox{3, 3, 3, 1, NoAbility},
+	SRBox{4, 3, 0, 0, DiscardHeld},
+	SRBox{3, 3, 4, 2, NoAbility},
+	SRBox{2, 4, 2, 0, ExtraAR},
+}
+
+func AdvanceSR(srTrack []SRBox, s *State, player Aff, roll int, ops int) (err error) {
+	playerPos := s.SpaceRace[player]
+	oppPos := s.SpaceRace[player.Opp()]
+
+	if playerPos >= len(srTrack) {
+		return fmt.Errorf("Player is already at the last Space Race position")
+	}
+
+	nextSRBox := srTrack[playerPos+1]
+	if ops < nextSRBox.OpsNeeded {
+		return fmt.Errorf("Player has insufficient ops to advance")
+	}
```

Comment:
suggest scraping these two checks out of this function into their own function.

I doubt they'll end up in here; this will be called only when the player has irrevocably decided to try to space a card, at which point they *must* have sufficient ops.